fvm/evm/precompiles/arch.go

```diff
@@ -48,9 +61,100 @@ func (c *flowBlockHeightFunction) Run(input []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	encoded := make([]byte, 8)
-	binary.BigEndian.PutUint64(encoded, bh)
-	// the EVM works natively in 256-bit words,
-	// we left pad to that size to prevent extra gas consumtion for masking.
-	return gethCommon.LeftPadBytes(encoded, 32), nil
+	// EVM works natively in 256-bit words,
+	// Encode to 256-bit is the common practice to prevent extra gas consumtion for masking.
+	buffer := make([]byte, EncodedUint64Size)
+	return buffer, EncodeUint64(bh, buffer, 0)
+}
+
+type proofVerifierFunction struct {
```

Comment:
I suggest renaming to  `proofVerifier`. Its not really a function.

Reply:
renamed it but they satisfy the Function interface, I might rename the function interface in the future to something else.